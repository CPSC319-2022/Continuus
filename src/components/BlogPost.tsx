import { MenuIcon } from "~/icons/Menu";

interface BlogPostProps {
  id: number;
  name: string;
  lastUpdated: string;
  title: string;
  content: string;
  imageUrl: string;
  comments: number;
}

export const BlogPost: React.FC<BlogPostProps> = ({
  id,
  name,
  lastUpdated,
  title,
  content,
  imageUrl,
  comments,
}) => {
  return (
    <div className="bg-base-150 card w-full rounded-md shadow-md shadow-slate-300">
      <div className="card-body m-[-15px]">
        <div className="mb-3 flex w-full justify-between">
          <div className="flex">
            <div className="avatar self-center">
              <div className="h-10 w-10 rounded-full">
                <img src={imageUrl} alt="avatar" />
              </div>
            </div>
            <div className="ml-3">
              <p className="text-lg font-bold">{name}</p>
              <p className="text-sm text-slate-400">{lastUpdated}</p>
            </div>
          </div>
          <div className="self-center">
            <div className="dropdown-left dropdown rounded-md shadow-slate-300">
              <button>
                <MenuIcon />
              </button>
              <ul
                tabIndex={0}
                className="dropdown-content menu rounded-box w-52 bg-base-100 p-2 shadow"
              >
                <li>
                  <a>Edit</a>
                </li>
                <li>
                  <a>Delete</a>
                </li>
              </ul>
            </div>
          </div>
        </div>
        <p className="mb-3 text-xl font-bold">{title}</p>
        <p>{content.length > 500 ? `${content.slice(0, 499)}...` : content}</p>
        <div className="self-end ">
          <label
            htmlFor={`modal-${id}`}
            className="btn-link text-highlight-green no-underline hover:cursor-pointer"
          >
            {comments} Comments
          </label>
        </div>
      </div>
    </div>
  );
};
