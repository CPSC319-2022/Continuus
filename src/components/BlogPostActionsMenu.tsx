import { MenuIcon } from "~/icons/Menu";
import { UpdateBlogPostWidget } from "~/components/update-blog-post-widget/UpdateBlogPostWidget";
import React from "react";
import { DeleteBlogPostWidget } from "~/components/delete-blog-post-widget/DeleteBlogPostWidget";

interface BlogPostActionsProps {
  id: string;
  title: string;
  content: string;
  isAuthor: boolean;
}

export const BlogPostActionsMenu: React.FC<BlogPostActionsProps> = ({id, title, content, isAuthor}) => {
  return (
    <div className="self-center">
      <div className="dropdown-left dropdown rounded-md shadow-slate-300">
        <button>
          <MenuIcon />
        </button>
        <ul
          tabIndex={0}
          className="dropdown-content menu rounded-box w-52 bg-base-100 p-2 shadow"
        >
          {
            isAuthor &&
              <li>
                  <UpdateBlogPostWidget id={id} title={title} content={content} />
              </li>
          }

          <li>
            <DeleteBlogPostWidget id={id} />
          </li>
        </ul>
      </div>
    </div>
  );
}