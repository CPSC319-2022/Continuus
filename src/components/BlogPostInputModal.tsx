import type { PropsWithChildren } from 'react';
import React from 'react';
import Modal from 'react-modal';

{process.env.NODE_ENV !== 'test' ? Modal.setAppElement("#__next") : Modal.setAppElement("body")}

export const BlogPostInputModal: React.FC<PropsWithChildren<{
    action: string,
    isOpen: boolean,
    closeDisabled?: boolean,
    onRequestClose: () => void,
}>> = ({
    children,
    action,
    isOpen,
    closeDisabled,
    onRequestClose,
}) => {
        return (
            <Modal
                isOpen={isOpen}
                onRequestClose={() => closeDisabled || onRequestClose()}
                contentLabel="Blog Post Input"
                overlayClassName="fixed inset-0 z-20 bg-black/75"
                ariaHideApp={process.env.NODE_ENV !== 'test'}
                className="bg-white absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 w-11/12 rounded-md md:w-1/2 z-40"
            >
                <div className="w-full h-12 flex flex-row justify-between items-center border-b border-b-gray-200 pl-4 pr-4">
                    <div className="font-semibold">{action} Blog Post</div>
                    {
                        closeDisabled || <button
                            className="rounded-full after:content-['\2715'] hover:text-emerald-400 transition-colors"
                            onClick={() => closeDisabled || onRequestClose()}
                        />
                    }
                </div>
                {children}
            </Modal>
        );
    }