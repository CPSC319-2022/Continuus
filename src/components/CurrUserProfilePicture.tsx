import { api } from "~/utils/api";
import { Spinner } from "./Spinner";

/**
 * @prop size: Width/Height of the Profile Picture in REM 
 * @returns 
 */
export const CurrUserProfilePicture: React.FC<{ size: number }> = ({ size }) => {
    const currUser = api.user.currentUser.useQuery();

    if (currUser.isLoading) {
        return <div className="aspect-square rounded-full" style={{ width: `${size}rem` }}>
            <Spinner size={size} />
        </div>
    }

    return (
        <img src={currUser.data?.image || "https://i.stack.imgur.com/34AD2.jpg"} alt="avatar" className="aspect-square rounded-full" style={{ width: `${size}rem` }} />

    )
}