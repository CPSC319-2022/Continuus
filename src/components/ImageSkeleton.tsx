/**
 * @prop size: Width/Height of the Skeleton in REM 
 * @returns 
 */
export const ImageSkeleton: React.FC<{ size: number }> = ({ size }) => {
    return (
        <div className="aspect-square rounded-full bg-gray-300 animate-pulse inline-block" style={{ width: `${size}rem` }} />
    );
}