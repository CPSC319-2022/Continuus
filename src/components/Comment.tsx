import { userPathToProfile } from "~/utils/profile";
import { timeAgo } from "~/utils/time";
import { ProfileName } from "./ProfileName";
import { ProfilePicture } from "./ProfilePicture";
import React, { useEffect, useState } from "react";
import { api } from "~/utils/api";
import { useSession } from "next-auth/react";
import { isAuthed, isAuthor } from "~/components/util";
import { Spinner } from "~/components/Spinner";

export interface CommentProps {
  dateAdded: Date;
  comment: string;
  name?: string | null;
  userId: string;
  imgUrl?: string | null;
  commentId: string;
  dateUpdated: Date;
}

export const Comment: React.FC<CommentProps> = ({
  dateAdded,
  comment,
  name,
  userId,
  imgUrl,
  commentId,
  dateUpdated,
}) => {
  const utils = api.useContext();
  const { status } = useSession();
  const [editInput, setEditInput] = useState<string>(comment);
  const [isEditing, setIsEditing] = useState(false);
  const [isDeletePending, setIsDeletePending] = useState(false);
  const [isDeleted, setIsDeleted] = useState(false);
  const currUser = api.user.currentUser.useQuery();

  const updateCommentMutation = api.comment.update.useMutation({
    onSuccess: () =>
      Promise.all([
        utils.comment.invalidate(),
        utils.blogPost.getOne.invalidate(),
        utils.blogPost.get.invalidate(),
      ]),
  });

  const deleteCommentMutation = api.comment.delete.useMutation({
    onSuccess: () =>
      Promise.all([
        utils.blogPost.getOne.invalidate(),
        utils.blogPost.get.invalidate(),
      ]),
  });

  useEffect(() => {
    if (updateCommentMutation.isSuccess) setIsEditing(false);
  }, [updateCommentMutation.isSuccess]);

  useEffect(() => {
    if (deleteCommentMutation.isSuccess) {
      setIsDeleted(true);
      setIsDeletePending(false);
    }
  }, [deleteCommentMutation.isSuccess]);

  const fetchedComment = api.comment.getUnique.useQuery({
    where: {
      id: commentId,
    },
  });

  if (fetchedComment.data !== undefined) comment = fetchedComment.data.content;

  const shouldSeeCommentActions = () => {
    return isAuthed(status) && isAuthor(currUser.data, userId);
  };

  const handleInputChange = (event: {
    target: { value: React.SetStateAction<string> };
  }) => {
    setEditInput(event.target.value);
  };

  const handleUpdateCancel = (event: React.MouseEvent<HTMLButtonElement>) => {
    event.preventDefault();
    setIsEditing(false);
    setEditInput(comment);
  };

  const handleCommentUpdate = () => {
    if (editInput) {
      updateCommentMutation.mutate({
        where: {
          id: commentId,
        },
        data: {
          content: editInput,
        },
      });
    }
  };

  const handleCommentDelete = () => {
    deleteCommentMutation.mutate({
      where: {
        id: commentId,
      },
    });
  };

  return (
    <>
      {isDeletePending ? (
        <div className="m-3 rounded-md bg-slate-100 p-4">
          <h1>Are you sure you want to delete this comment?</h1>
          <button
            className="ml-2 mt-3 rounded bg-red-600 py-0.5 px-2 text-white hover:bg-red-800"
            onClick={() => handleCommentDelete()}
          >
            Delete
          </button>
          <button
            className="ml-2 mt-3 rounded bg-gray-300 py-0.5 px-2 text-gray-800 hover:bg-gray-400"
            onClick={(event) => {
              event.preventDefault();
              setIsDeletePending(false);
            }}
          >
            Cancel
          </button>
        </div>
      ) : (
        !isDeleted && (
          <div className="m-3 rounded-md border-b p-4">
            <div className="flex">
              <div className="avatar self-center">
                <ProfilePicture
                  size={2.5}
                  redirectLink={userPathToProfile(userId)}
                  imgUrl={imgUrl}
                />
              </div>
              <div className="ml-3">
                <ProfileName name={name || ""} userId={userId} />
                <p className="text-sm tex-gray-400">
                  {timeAgo(dateAdded)}
                  {dateAdded.getTime() !== dateUpdated.getTime()
                    ? ` (updated ${timeAgo(dateUpdated)})`
                    : ""}
                </p>
              </div>
              {shouldSeeCommentActions() && (
                <div className="ml-auto">
                  <button
                    className="rounded-xl bg-green-600 p-1.5 text-white transition-all duration-300 hover:rounded-3xl hover:bg-green-800"
                    onClick={() => {
                      setIsEditing(true);
                    }}
                  >
                    <svg
                      xmlns="http://www.w3.org/2000/svg"
                      className="h-5 w-5"
                      fill="none"
                      viewBox="0 0 24 24"
                      stroke="currentColor"
                      stroke-width="2"
                    >
                      <path
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        d="M11 5H6a2 2 0 00-2 2v11a2 2 0 002 2h11a2 2 0 002-2v-5m-1.414-9.414a2 2 0 112.828 2.828L11.828 15H9v-2.828l8.586-8.586z"
                      />
                    </svg>
                  </button>
                  <button
                    className="ml-3 mr-2 rounded-xl bg-red-700 p-1.5 text-white transition-all duration-300 hover:rounded-3xl hover:bg-red-900"
                    onClick={() => setIsDeletePending(true)}
                  >
                    <svg
                      xmlns="http://www.w3.org/2000/svg"
                      fill="currentColor"
                      className="bi bi-trash h-5 w-5"
                      viewBox="0 0 16 16"
                    >
                      <path
                        d="M5.5 5.5A.5.5 0 0 1 6 6v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm2.5 0a.5.5 0 0 1 .5.5v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm3 .5a.5.5 0 0 0-1 0v6a.5.5 0 0 0 1 0V6z"
                        fill="white"
                      ></path>
                      <path
                        fill-rule="evenodd"
                        d="M14.5 3a1 1 0 0 1-1 1H13v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V4h-.5a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1H6a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1h3.5a1 1 0 0 1 1 1v1zM4.118 4 4 4.059V13a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1V4.059L11.882 4H4.118zM2.5 3V2h11v1h-11z"
                        fill="white"
                      ></path>
                    </svg>
                  </button>
                </div>
              )}
            </div>
            {updateCommentMutation.isLoading ? (
              <div className="flex w-full items-center justify-center">
                <Spinner size={2} />
              </div>
            ) : isEditing ? (
              <div className="mt-3 p-2">
                <input
                  type="text"
                  value={editInput}
                  onChange={handleInputChange}
                  className="border-2 border-slate-500"
                />
                <button
                  className={`ml-2 rounded bg-gray-300 py-0.5 px-2 text-gray-800 ${
                    editInput ? "hover:bg-gray-400" : "cursor-not-allowed"
                  }`}
                  onClick={() => handleCommentUpdate()}
                >
                  Save
                </button>
                <button
                  className="ml-2 rounded bg-gray-300 py-0.5 px-2 text-gray-800 hover:bg-gray-400"
                  onClick={(event) => handleUpdateCancel(event)}
                >
                  Cancel
                </button>
              </div>
            ) : (
              <div className="break-words">
                <p className="mt-4 w-full">{comment}</p>
              </div>
            )}
          </div>
        )
      )}
    </>
  );
};
