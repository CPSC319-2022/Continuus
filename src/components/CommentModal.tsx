import { Comment } from "./Comment";
import React, { useState } from "react";
import ReactMarkdown from "react-markdown";
import remarkGfm from "remark-gfm";
import remarkSlug from "remark-slug";
import type { Comment as CommentType, User } from "@prisma/client";
import { timeAgo } from "~/utils/time";
import { ProfilePicture } from "./ProfilePicture";
import { BlogPostActionsMenu } from "~/components/BlogPostActionsMenu";
import { api } from "~/utils/api";
import { useSession } from "next-auth/react";
import { shouldSeeActions, isAuthor } from "~/components/util";
import {userPathToProfile} from "~/utils/profile";

type CommentEntry = CommentType & { user: User };

export interface CommentModalProps {
  id: string;
  title: string;
  poster: string;
  lastUpdated: Date;
  createdAt: Date;
  post: string;
  comments: CommentEntry[];
  content: string;
  authorId: string;
  imgUrl?: string | null;
}

export const CommentModal: React.FC<CommentModalProps> = ({
  id,
  title,
  poster,
  lastUpdated,
  createdAt,
  post,
  comments,
  content,
  authorId,
  imgUrl,
}) => {
  const [input, setInput] = useState<string>("");

  const utils = api.useContext();

  const currUser = api.user.currentUser.useQuery();

  const { status } = useSession();

  const createCommentMutation = api.comment.create.useMutation({
    onSuccess() {
        return Promise.all([utils.blogPost.get.invalidate(),
                            utils.comment.count.invalidate()]);
    },
  });

  const userId = currUser?.data?.id;

  const handleCommentChange = (event: {
    target: { value: React.SetStateAction<string> };
  }) => {
    setInput(event.target.value);
  };

  const handlePostButtonClick = () => {
    createCommentMutation.mutate({
      data: {
        userId: userId!,
        blogPostId: id,
        content: input,
      },
    });

    setInput("");
  };

  return (
    <>
      <input type="checkbox" id={`modal-${id}`} className="modal-toggle" />
      <label htmlFor={`modal-${id}`} className="modal z-10 cursor-pointer">
        <label
          className="card-body modal-box relative m-[-10px] w-11/12 max-w-5xl rounded-md"
          htmlFor=""
        >
          <div className="mb-3 flex w-full justify-between">
            <div className="flex">
              <div className="avatar self-center">
                  <ProfilePicture 
                      size={2.5}
                      redirectLink={userPathToProfile(authorId)} 
                      imgUrl={imgUrl}
                  />
              </div>
              <div className="ml-3">
                <p className="text-lg font-bold">{poster}</p>
                <p className="text-sm text-slate-400">{`${timeAgo(createdAt)}${
                  createdAt.getTime() !== lastUpdated.getTime()
                    ? ` (updated ${timeAgo(lastUpdated)})`
                    : ""
                }`}</p>
              </div>
            </div>
            {
              shouldSeeActions(status, currUser.data, authorId) && (
              <BlogPostActionsMenu
                id={id}
                title={title}
                content={content}
                isAuthor={isAuthor(currUser.data, authorId)}
              />
            )}
          </div>
          <p className="mb-3 text-xl font-bold">{title}</p>
          <div className="prose max-w-none ">
            <ReactMarkdown remarkPlugins={[remarkGfm, remarkSlug]}>
              {post}
            </ReactMarkdown>
          </div>
          <div className="self-end">
            <p className="btn-link text-highlight-green no-underline hover:no-underline">
              {comments.length} Comments
            </p>
          </div>
          <div>
            {comments.map(
              ({
                id: commentId,
                content: comment,
                createdAt: dateAdded,
                user: { name: authorName, id: commenterId, image},
              }) => (
                <Comment
                  key={`${commentId}`}
                  dateAdded={dateAdded}
                  comment={comment}
                  name={authorName}
                  userId={commenterId}
                  imgUrl={image}
                />
              )
            )}
          </div>
          <div className="m-[2rem] mx-0 mb-0">
            <div
              className={`${userId ? "" : "tooltip tooltip-info"} w-full`}
              data-tip="Sign in to comment"
            >
              <input
                type="text"
                placeholder="Write a new comment"
                value={input}
                onChange={handleCommentChange}
                readOnly={userId ? false : true}
                disabled={userId ? false : true}
                className={`${
                  userId ? "" : "input-disabled cursor-not-allowed"
                }input-bordered mb-1 w-full rounded-sm border-[1px] p-2`}
              />
            </div>

            <div className="mb-[-10px] flex justify-end gap-2">
              <button
                className={`${
                  userId
                    ? " border-highlight-green bg-highlight-green "
                    : "input-disabled cursor-not-allowed bg-gray-400 text-black"
                } mt-[0.5rem] h-10 rounded-sm px-5`}
                onClick={handlePostButtonClick}
                disabled={userId ? false : true}
              >
                Post
              </button>
              <label
                className={`mt-[0.5rem] h-10 rounded-sm border-[2px] border-slate-200 bg-white px-5 pt-[0.38rem] text-center text-black hover:cursor-pointer`}
                htmlFor={`modal-${id}`}
              >
                Cancel
              </label>
            </div>
          </div>
        </label>
      </label>
    </>
  );
};
