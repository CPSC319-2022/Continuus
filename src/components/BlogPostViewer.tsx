import type { BlogPost, User, Comment } from "@prisma/client";
import { useState, useMemo, useEffect } from "react";
import { useInView } from "react-intersection-observer";
import { api } from "~/utils/api";
import { CommentModal } from "./CommentModal";
import { Spinner } from "./Spinner";
import { BlogPost as BlogPostComponent } from "./BlogPost";

type Post = BlogPost & { user: User; comments: (Comment & { user: User })[] };

export const BlogPostViewer: React.FC = () => {
  const [view, setView] = useState<string>("Recent");
  const {
    data: blogPosts,
    fetchNextPage,
    hasNextPage,
  } = api.blogPost.get.useInfiniteQuery(
    {
      take: 20,
    },
    { getNextPageParam: (lastPage) => ({ id: lastPage.nextCursor }) }
  );

  const posts = useMemo(
    () =>
      blogPosts?.pages.reduce(
        (result, curr) => [...result, ...curr.items],
        [] as Post[]
      ),
    [blogPosts]
  );

  const { ref, inView } = useInView();

  useEffect(() => {
    if (inView) {
      void (async () => {
        await fetchNextPage();
      })();
    }
  }, [inView, fetchNextPage]);

  return (
    <div className="w-full md:w-1/2 ">
      <div className="mb-6 flex w-full justify-end">
        <select
          className="h-8 w-64 max-w-xs border-b-2 bg-white"
          onChange={(e) => setView(e.target.value)}
        >
          <option>Recent</option>
          <option>Popular</option>
        </select>
      </div>
      {posts?.length === 0
        ? "Nothing to see here"
        : posts?.map(
            ({
              id,
              title,
              updatedAt,
              content,
              comments,
              user: { name, image },
            }) => (
              <>
                <div key={id} className="mb-6">
                  <BlogPostComponent
                    id={id}
                    name={name as string}
                    title={title}
                    lastUpdated={updatedAt}
                    imageUrl={image as string}
                    content={content}
                    comments={(comments as Comment[]).length}
                  />
                </div>
                <CommentModal
                  id={id}
                  title={title}
                  comments={comments as (Comment & { user: User })[]}
                  poster={name as string}
                  lastUpdated={updatedAt}
                  post={content}
                  posterAvatarUrl={image as string}
                />
              </>
            )
          )}
      {hasNextPage && (
        <div className="my-5 flex justify-center" ref={ref}>
          <Spinner size={2} />
        </div>
      )}
    </div>
  );
};
