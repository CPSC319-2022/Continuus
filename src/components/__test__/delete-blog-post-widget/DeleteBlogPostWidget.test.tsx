import { describe, it, expect, beforeEach, vi } from "vitest";
import { cleanup, render, screen } from "@testing-library/react";
import { CommentProps } from "~/components/Comment";
import userEvent from "@testing-library/user-event";

const setup = (isSuccess: boolean) => {
  vi.mock("~/utils/api", () => ({
    api: {
      blogPost: {
        delete: {
          useMutation: () => ({
            isSuccess: false,
            isLoading: false,
          }),
        },
      },
      user: {
        currentUser: {
          useQuery: () => ({
            data: null,
            isLoading: false,
          }),
        },
      },
    },
  }));
};

describe("Delete blog post widget: Snapshot", () => {
  beforeEach(() => {
    cleanup();
    vi.clearAllMocks();
    vi.resetModules();
  });

  it("render component", async () => {
    setup(false);

    const { DeleteBlogPostWidget } = await import(
      "~/components/delete-blog-post-widget/DeleteBlogPostWidget"
    );

    const body = render(<DeleteBlogPostWidget id="test-id" />).baseElement;
    expect(body).toMatchSnapshot();
  });

  it("open widget", async () => {
    setup(false);

    const { DeleteBlogPostWidget } = await import(
      "~/components/delete-blog-post-widget/DeleteBlogPostWidget"
    );

    const body = render(<DeleteBlogPostWidget id="test-id" />);
    await userEvent.click(screen.getByTestId("delete-blog-post-button"));
    expect(body.baseElement).toMatchSnapshot();
  });

  it("delete successful", async () => {
    setup(true);

    const { DeleteBlogPostWidget } = await import(
      "~/components/delete-blog-post-widget/DeleteBlogPostWidget"
    );

    const body = render(<DeleteBlogPostWidget id="test-id" />);
    await userEvent.click(screen.getByTestId("delete-blog-post-button"));
    expect(body.baseElement).toMatchSnapshot();
  });
});
