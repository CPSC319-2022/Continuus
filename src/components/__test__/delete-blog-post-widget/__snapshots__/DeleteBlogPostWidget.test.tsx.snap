// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Delete blog post widget: Snapshot > delete successful 1`] = `
<body
  class="ReactModal__Body--open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    class="ReactModalPortal"
  />
  <div>
    <div
      data-testid="delete-blog-post-button"
    >
      <span>
        Delete
      </span>
    </div>
  </div>
  <div
    class="ReactModalPortal"
  >
    <div
      class="ReactModal__Overlay fixed inset-0 z-20 bg-black/75"
    >
      <div
        aria-label="Blog Post Input"
        aria-modal="true"
        class="ReactModal__Content absolute top-1/2 left-1/2 z-40 w-11/12 -translate-x-1/2 -translate-y-1/2 rounded-md bg-white md:w-1/2 p-4"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="flex pb-3 w-full flex-row items-center justify-between border-b border-b-gray-200 pl-4 pr-4"
        >
          <div
            class="font-semibold"
          >
            Delete
             Blog Post
          </div>
          <button
            class="rounded-full transition-colors after:content-['\\\\2715'] hover:text-highlight-red hover:scale-125 transition-all"
          />
        </div>
        <div
          class="flex h-24 w-full items-center justify-center"
        >
          <div
            class="text-lg"
          >
            Are you sure you want to delete this post?
          </div>
        </div>
        <form>
          <input
            class="relative left-full -translate-x-full h-10 rounded-md border bg-highlight-red w-32 text-center text-white hover:cursor-pointer hover:bg-gray-700 transition-all"
            type="submit"
            value="Delete"
          />
        </form>
      </div>
    </div>
  </div>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
</body>
`;

exports[`Delete blog post widget: Snapshot > open widget 1`] = `
<body
  class="ReactModal__Body--open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div>
    <div
      data-testid="delete-blog-post-button"
    >
      <span>
        Delete
      </span>
    </div>
  </div>
  <div
    class="ReactModalPortal"
  >
    <div
      class="ReactModal__Overlay fixed inset-0 z-20 bg-black/75"
    >
      <div
        aria-label="Blog Post Input"
        aria-modal="true"
        class="ReactModal__Content absolute top-1/2 left-1/2 z-40 w-11/12 -translate-x-1/2 -translate-y-1/2 rounded-md bg-white md:w-1/2 p-4"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="flex pb-3 w-full flex-row items-center justify-between border-b border-b-gray-200 pl-4 pr-4"
        >
          <div
            class="font-semibold"
          >
            Delete
             Blog Post
          </div>
          <button
            class="rounded-full transition-colors after:content-['\\\\2715'] hover:text-highlight-red hover:scale-125 transition-all"
          />
        </div>
        <div
          class="flex h-24 w-full items-center justify-center"
        >
          <div
            class="text-lg"
          >
            Are you sure you want to delete this post?
          </div>
        </div>
        <form>
          <input
            class="relative left-full -translate-x-full h-10 rounded-md border bg-highlight-red w-32 text-center text-white hover:cursor-pointer hover:bg-gray-700 transition-all"
            type="submit"
            value="Delete"
          />
        </form>
      </div>
    </div>
  </div>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
</body>
`;

exports[`Delete blog post widget: Snapshot > render component 1`] = `
<body>
  <div>
    <div
      data-testid="delete-blog-post-button"
    >
      <span>
        Delete
      </span>
    </div>
  </div>
  <div
    class="ReactModalPortal"
  />
</body>
`;
